File: app/src/main/java/me/kashyap/theverge/AppComponent.java
package me.kashyap.theverge;

import android.app.Activity;
import android.app.Application;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created on 3/26/2015.
 */
@Singleton
@Component(modules = {AppModule.class, UiModule.class})
public interface AppComponent {

    void inject(MainActivity activity);
    void inject(Application application);

    final class Initializer {
        static AppComponent init(RssApplication rssApplication) {
            return Dagger_AppComponent
                    .builder()
                    .appModule(new AppModule(rssApplication))
                    .build();

        }
    }

}

File: app/src/main/kotlin/me/kashyap/theverge/AppComponent.kt
package me.kashyap.theverge

import dagger.Component
import javax.inject.Singleton

/**
 * Created on 3/26/2015.
 */
@Singleton
@Component(modules = arrayOf(AppModule::class, UiModule::class))
interface AppComponent {

    fun inject(activity: MainActivity)
    fun inject(application: RssApplication)

    object Initializer {
        internal fun init(rssApplication: RssApplication): AppComponent {
            return DaggerAppComponent.builder().appModule(AppModule(rssApplication)).build()

        }
    }

}
